packages/sync-actions/src/utils/action-map-custom.js

```diff
@@ -24,50 +24,43 @@ const extractTypeFields = (diffedFields, nextFields) =>
   Array.isArray(diffedFields)
     ? diffpatcher.getDeltaValue(diffedFields)
     : nextFields
-const extractFieldValue = (diffedFields, nextFields, fieldName) =>
-  Array.isArray(diffedFields[fieldName])
-    ? diffpatcher.getDeltaValue(diffedFields[fieldName])
-    : nextFields[fieldName]
+const extractFieldValue = (newFields, fieldName) => newFields[fieldName]
```

Comment:
So this in essence is the change. The rest is a bit of renaming?

Reply:
Yeah

Reply:
So what does the diffpatcher return if it is an array? Is is something else than the `newFields[fieldName]`? Did we check that?

Reply:
The main change is that we don't call `diffpatcher.getDeltaValue()` anymore, because we already now that we changed a text/string field. And `diffpatcher.getDeltaValue()` tries to patch the old field with the delta, which for me doesn't make a lot of sense if you already have the `newFields[fieldName]. Why should I then complicate things by applying a patch to the old value?

Reply:
@tdeekens I will revert the changes to our `yarn.lock` and `TODOS.md`, looks like I didn't check my changes properly before opening this PR 😞

Reply:
> And diffpatcher.getDeltaValue() tries to patch the old field with the delta, which for me doesn't make a lot of sense if you already have the `newFields[fieldName].

To me neither. I was just wondering if there is an expectation towards that somewhere. But I also can't recall. I guess we have to try :)

Reply:
a code comment outlining that `fields` of `custom` is not an array would be helpful and that they are of a different shape to `fieldDefinitions` (which is an array)